lua/telescope/builtin/lsp.lua

```diff
@@ -120,22 +120,24 @@ lsp.code_actions = function(opts)
     return
   end
 
-  local _, response = next(results_lsp)
-  if not response then
-    print("No code actions available")
-    return
+  local idx = 1
+  local results = {}
+
+  for _, response in pairs(results_lsp) do
+    if response.result then
+      for _, result in pairs(response.result) do
+        result.idx = idx
+        idx = idx + 1
+        table.insert(results, result)
```

Comment:
we could switch those two lines and basically do
```lua
results[idx] = result
idx = idx + 1
```

do you know why `idx` is part of the table? Tbh i never looked at that code :)

Reply:
It's for the entry maker: https://github.com/nvim-telescope/telescope.nvim/blob/b3fdbbdcd09a9f704274910a900ee852a7b16859/lua/telescope/builtin/lsp.lua#L149-L150

I personally find prefixing the display with the idx pretty pointless though tbh

Reply:
Oh right i see. Lets keep the index that way i can input `4` and jump to the fourth entry

Reply:
> Oh right i see. Lets keep the index that way i can input `4` and jump to the fourth entry

This doesn't seem to ever work for me though, otherwise I'd agree it'd be very helpful. https://asciinema.org/a/nthue7KhvwMAJiionPjA7CtSg

Reply:
Looks like a issue with the ngram sorter(default) (doesn't do anything with prompt < 2) try fzy.

Reply:
Ah. I had issues with getting fzy to work due to being on an M1 mac :(

Reply:
Really? I was under the impression that it worked.

*(cancels his orders for an M1 Macbook)*

Reply:
@clason :laughing: 
@williamboman We also have a lua version, no fzy-native so not as fast but still better sorting (imo)
```lua
file_sorter =  require'telescope.sorters'.get_fzy_sorter,
generic_sorter =  require'telescope.sorters'.get_fzy_sorter,
```

Reply:
No! I want fast! 😠 

:D

Reply:
> Really? I was under the impression that it worked.

Yeah still getting this:

```
Error in packer_compiled: Vim(lua):E5108: Error executing lua [string ":lua"]:15: .../start/telescope
.nvim/lua/telescope/_extensions/init.lua:11: This extension doesn't exist or is not installed: fzy_na
tive^@...scope-fzy-native.nvim/deps/fzy-lua-native/lua/native.lua:34: dlopen(/Users/williamboman/.loc
al/share/nvim/site/pack/packer/start/telescope-fzy-native.nvim/deps/fzy-lua-native/lua/../static/libf
zy-darwin-arm64.so, 5): image not found
```

Reply:
`fzy-native` should be possible, just not magically because of missing CI machines for m1 s. Had a very one sided conversation here :rofl: 
https://github.com/nvim-telescope/telescope-fzy-native.nvim/issues/4#issuecomment-790059295

So my guess would be:
- locally make
- add the tbl value so it gets loaded

Reply:
https://github.com/romgrk/fzy-lua-native/issues/13

Reply:
Ah, so you can trick local runners into producing native M1 binaries, but there are no such GH runners (yet). Bummer.

Reply:
> * locally make

This was enough actually. 

```
$ cd ~/.local/share/nvim/site/pack/packer/start/telescope-fzy-native.nvim/deps/fzy-lua-native/ && make
```

Reply:
Great, thanks! In that case I could just add `make` to the packer config for `fzy-native` and have it update automatically.